bin/compile_dependencies.sh

```diff
@@ -38,5 +38,6 @@ pip-compile \
     --no-emit-index-url \
     --output-file requirements/dev.txt \
     "$@" \
-    requirements/ci.txt \
+    requirements/base.txt \
```

Comment:
Why is this change needed? I thought it would be faster to reuse already compiled ci.txt here?

Reply:
We could revert this and include `pytest` in the dev requirements for other projects :thinking:  (to keep the requirements similar to the other projects). `pytest` is needed for projects which need to generate documentation through github actions.